// node-opcua-service-translate-browse-path/dist/imports.js
"use strict";
Object.defineProperty(exports, "__esModule", { value: true });
/**
 * @module node-opcua-service-translate-browse-path
 */
var node_opcua_types_1 = require("node-opcua-types");
Object.defineProperty(exports, "RelativePathElement", { enumerable: true, get: function () { return node_opcua_types_1.RelativePathElement; } });
Object.defineProperty(exports, "RelativePath", { enumerable: true, get: function () { return node_opcua_types_1.RelativePath; } });
Object.defineProperty(exports, "BrowsePath", { enumerable: true, get: function () { return node_opcua_types_1.BrowsePath; } });
Object.defineProperty(exports, "TranslateBrowsePathsToNodeIdsRequest", { enumerable: true, get: function () { return node_opcua_types_1.TranslateBrowsePathsToNodeIdsRequest; } });
Object.defineProperty(exports, "BrowsePathTarget", { enumerable: true, get: function () { return node_opcua_types_1.BrowsePathTarget; } });
Object.defineProperty(exports, "BrowsePathResult", { enumerable: true, get: function () { return node_opcua_types_1.BrowsePathResult; } });
Object.defineProperty(exports, "TranslateBrowsePathsToNodeIdsResponse", { enumerable: true, get: function () { return node_opcua_types_1.TranslateBrowsePathsToNodeIdsResponse; } });
